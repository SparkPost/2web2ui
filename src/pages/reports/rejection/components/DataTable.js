import React, { Component } from 'react';
import { TableCollection, Empty, LongTextContainer } from 'src/components';
import { Panel } from 'src/components/matchbox';
import AddFilterLink from '../../components/AddFilterLink';

const columns = [
  { label: 'Reason', width: '45%', sortKey: 'reason' },
  { label: 'Domain', sortKey: 'domain' },
  { label: 'Category', sortKey: 'rejection_category_name' },
  { label: 'Count', sortKey: 'count_rejected' },
];

export class DataTable extends Component {
  getRowData = rowData => {
    const { reason, domain, rejection_category_name, count_rejected } = rowData;
    return [
      <LongTextContainer text={reason} />,
      <AddFilterLink
        newFilter={{ type: 'Recipient Domain', value: domain }}
        reportType="rejections"
        content={domain}
      />,
      rejection_category_name,
      count_rejected,
    ];
  };

  render() {
    const { list } = this.props;

    if (!list.length) {
      return (
        <Panel.LEGACY>
          <Empty message="No rejection reasons to report" />
        </Panel.LEGACY>
      );
    }

    return (
      <TableCollection
        columns={columns}
        rows={list}
        getRowData={this.getRowData}
        pagination
        defaultSortColumn="reason"
        defaultSortDirection="desc"
        filterBox={{
          show: true,
          keyMap: { category: 'rejection_category_name' },
          itemToStringKeys: ['rejection_category_name', 'domain', 'reason'],
          exampleModifiers: ['domain', 'category'],
          matchThreshold: 5,
        }}
      />
    );
  }
}

export default DataTable;
