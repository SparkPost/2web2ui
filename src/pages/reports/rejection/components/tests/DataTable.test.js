import React from 'react';
import { DataTable } from '../DataTable';
import { shallow } from 'enzyme';

jest.mock('src/helpers/reports');

describe('Rejection Data Table: ', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      list: [
        {
          count_rejected: 65,
          rejection_category_name: 'Policy Rejection',
          rejection_category_id: 1,
          reason: '551 - Cannot Relay 105',
          domain: 'gmail.com'
        },
        {
          count_rejected: 62,
          rejection_category_name: 'Policy Rejection',
          rejection_category_id: 1,
          reason: '550 - Invalid recipient ...@... (#5.1.1)',
          domain: 'gmail.com'
        }
      ],
      addFilters: jest.fn()
    };
    wrapper = shallow(<DataTable {...props} />);
  });

  afterEach(() => { jest.resetAllMocks(); });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with no rejections', () => {
    wrapper.setProps({ list: []});
    expect(wrapper).toMatchSnapshot();
  });

  it('getRowData should render row data properly', () => {
    const rows = wrapper.instance().getRowData({ reason: 'bad delay', rejection_category_name: 'cat1', count_rejected: 10, domain: 'gmail.com' });
    expect(rows).toMatchSnapshot();
  });

});
