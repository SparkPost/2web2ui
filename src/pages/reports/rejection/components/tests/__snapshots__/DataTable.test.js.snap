// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rejection Data Table:  getRowData should render row data properly 1`] = `
Array [
  <LongTextContainer
    text="bad delay"
  />,
  <Memo(Connect(AddFilterLink))
    content="gmail.com"
    newFilter={
      Object {
        "type": "Recipient Domain",
        "value": "gmail.com",
      }
    }
    reportType="rejections"
  />,
  "cat1",
  10,
]
`;

exports[`Rejection Data Table:  renders correctly 1`] = `
<TableCollection
  columns={
    Array [
      Object {
        "label": "Reason",
        "sortKey": "reason",
        "width": "45%",
      },
      Object {
        "label": "Domain",
        "sortKey": "domain",
      },
      Object {
        "label": "Category",
        "sortKey": "rejection_category_name",
      },
      Object {
        "label": "Count",
        "sortKey": "count_rejected",
      },
    ]
  }
  defaultSortColumn="reason"
  defaultSortDirection="desc"
  filterBox={
    Object {
      "exampleModifiers": Array [
        "domain",
        "category",
      ],
      "itemToStringKeys": Array [
        "rejection_category_name",
        "domain",
        "reason",
      ],
      "keyMap": Object {
        "category": "rejection_category_name",
      },
      "matchThreshold": 5,
      "show": true,
    }
  }
  getRowData={[Function]}
  pagination={true}
  rows={
    Array [
      Object {
        "count_rejected": 65,
        "domain": "gmail.com",
        "reason": "551 - Cannot Relay 105",
        "rejection_category_id": 1,
        "rejection_category_name": "Policy Rejection",
      },
      Object {
        "count_rejected": 62,
        "domain": "gmail.com",
        "reason": "550 - Invalid recipient ...@... (#5.1.1)",
        "rejection_category_id": 1,
        "rejection_category_name": "Policy Rejection",
      },
    ]
  }
/>
`;

exports[`Rejection Data Table:  renders correctly with no rejections 1`] = `
<Panel.LEGACY>
  <Empty
    message="No rejection reasons to report"
  />
</Panel.LEGACY>
`;
